src/it/resources/com/google/checkstyle/test/chapter5naming/rule523methodnames/InputMethodName.java

```diff
@@ -60,17 +87,47 @@ void Testing_Foo2() {} // violation 'Method name 'Testing_Foo2' must match patte
   @Test
   void TestingFooBad() {} // violation 'Method name 'TestingFooBad' must match pattern'
 
+  @Test
+  void testing_foo_() {} // violation 'Method name 'testing_foo_' must match pattern'
+
+  @Test
+  void testing_Foo_() {} // violation 'Method name 'testing_Foo_' must match pattern'
+
+  @Test
+  void testing__foo() {} // violation 'Method name 'testing__foo' must match pattern'
+
+  @Test
+  void testing__Foo() {} // violation 'Method name 'testing__Foo' must match pattern'
+
+  @Test
+  void testing__foo_() {} // violation 'Method name 'testing__foo_' must match pattern'
+
+  @Test
+  void testing__Foo_() {} // violation 'Method name 'testing__Foo_' must match pattern'
```

Comment:
can you also add test cases for digits right after consecutive underscores? and also trailing underscore? like 

```
testing__09() {}
testing_09_() {}
testing__09_() {}
testing__09__() {}
```